pkg/containers/service.go

```diff
@@ -0,0 +1,35 @@
+package containers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aquasecurity/tracee/pkg/containers/runtime"
+)
+
+type runtimeInfoService struct {
+	enrichers map[string]runtime.ContainerEnricher
+}
+
+func RuntimeInfoService() runtimeInfoService {
+	return runtimeInfoService{
+		enrichers: make(map[string]runtime.ContainerEnricher),
+	}
+}
+
+func (e *runtimeInfoService) Register(runtime string, socket string, enricherBuilder func(socket string) (runtime.ContainerEnricher, error)) error {
```

Comment:
what if a user calls more than once register? we should protect it inside the function (not allowing multiple registering)

Reply:
Sounds like an override? I don't think it's an inappropriate use of the API.